lib/cyclonedx/report.rb

```diff
@@ -53,6 +53,8 @@ def to_cyclonedx
               "with error message #{e.class}: #{e.message}"
         bugsnag_notify(msg)
       end
+
+      cyclonedx_report[:components].uniq!
```

Comment:
I'm not sure uniq will work on an array of hashes
```
[{'a'=>1,'b'=>2},{'a'=>1.5, 'b'=>2}].uniq
```
=> [{"a"=>1, "b"=>2}, {"a"=>1.5, "b"=>2}]

You may need to roll with something similar to
```
cyclonedx_report[:components] = cyclonedx_report[:components].inject([]) do |result,h|
   result << h unless result.include?(h); result
end
```